### PR TITLE
Correct resuming of fully interacting state and don't generate u_n.out file in analyze

### LINKED ISSUE #520: MBAR analysis: Actual column sum for state 0 was 4731.000000
After running calculations with the new scheme for the fully interacting energy, the analysis gives:

```
Traceback (most recent call last):
  File "/cbio/jclab/home/andrrizzi/miniconda/bin/yank", line 11, in <module>
    load_entry_point('yank==0.11.1', 'console_scripts', 'yank')()
  File "/cbio/jclab/home/andrrizzi/miniconda/lib/python2.7/site-packages/yank-0.11.1-py2.7-linux-x86_64.egg/yank/cli.py", line 74, in main
    dispatched = getattr(commands, command).dispatch(command_args)
  File "/cbio/jclab/home/andrrizzi/miniconda/lib/python2.7/site-packages/yank-0.11.1-py2.7-linux-x86_64.egg/yank/commands/analyze.py", line 63, in dispatch
    analyze.analyze(args['--store'])
  File "/cbio/jclab/home/andrrizzi/miniconda/lib/python2.7/site-packages/yank-0.11.1-py2.7-linux-x86_64.egg/yank/analyze.py", line 502, in analyze
    (Deltaf_ij, dDeltaf_ij) = estimate_free_energies(ncfile, mbar = mbar)
  File "/cbio/jclab/home/andrrizzi/miniconda/lib/python2.7/site-packages/yank-0.11.1-py2.7-linux-x86_64.egg/yank/analyze.py", line 270, in estimate_free_energies
    (Deltaf_ij, dDeltaf_ij) = mbar.getFreeEnergyDifferences()
  File "/cbio/jclab/home/andrrizzi/miniconda/lib/python2.7/site-packages/pymbar-2.1.0b0-py2.7-linux-x86_64.egg/pymbar/mbar.py", line 410, in getFreeEnergyDifferences
  File "/cbio/jclab/home/andrrizzi/miniconda/lib/python2.7/site-packages/pymbar-2.1.0b0-py2.7-linux-x86_64.egg/pymbar/mbar.py", line 1665, in _computeAsymptoticCovarianceMatrix
pymbar.utils.ParameterError: Warning: Should have \sum_n W_nk = 1.  Actual column sum for state 0 was 46930.000000. 1 other columns have similar problems
```

### PR DESCRIPTION
**Pull request by andrrizzi**
Fix #520 and #505 . This is not ready yet. I want to add a short test for this bug. Should I bump version?

### CODE REVIEW

File: Yank/sampling.py [hunk 1]
```diff
@@ -382,10 +382,7 @@ def _cache_context(self):
         state = self.states[0]
         self._integrator = openmm.LangevinIntegrator(state.temperature, self.collision_rate, self.timestep)
         self._integrator.setRandomNumberSeed(int(np.random.randint(0, MAX_SEED)))
-        if self.platform:
-            self._context = openmm.Context(state.system, self._integrator, self.platform)
-        else:
-            self._context = openmm.Context(state.system, self._integrator)
+        self._context = self._create_context(state.system, self._integrator)
```
Comment (Lnaden) [inline Yank/sampling.py]: Is there a reason you don't want to allow a platform to be specified here?

File: Yank/sampling.py [hunk 2]
```diff
@@ -402,14 +399,13 @@ def _cache_context(self):
...
-                self._reference_context = openmm.Context(reference_state.system, reference_state_integrator)
-                self._reference_LJ_context = openmm.Context(reference_LJ_state.system, reference_LJ_state_integrator)
-                self._reference_LJ_expanded_context = openmm.Context(reference_LJ_expanded_state.system, reference_LJ_expanded_state_integrator)
+
+            self._reference_context = self._create_context(reference_state.system,
+                                                           reference_state_integrator)
+            self._reference_LJ_context = self._create_context(reference_LJ_state.system,
+                                                              reference_LJ_state_integrator)
+            self._reference_LJ_expanded_context = self._create_context(reference_LJ_expanded_state.system,
+                                                                       reference_LJ_expanded_state_integrator)
```
Comment (Lnaden) [inline Yank/sampling.py]: Again, why not let a platform be chosen?

File: Yank/sampling.py (hunk 1 above)
Comment (andrrizzi) [inline Yank/sampling.py]: `ReplicaExchange._create_context()` is a shortcut function that implements the
```python
if self.platform:
   return Context(..., self.platform)
else:
   return Context(...)
```

Comment (Lnaden) [inline Yank/sampling.py]: Ah, okay, I see it now

**Comment by Lnaden** [comment]
Go ahead and update the minor version number to 0.11.1

**Comment by Lnaden** [comment]
err.. 0.11.2

**Comment by andrrizzi** [comment]
Test pass! If the `setup.py` file is correct, this is ready to be merged.

**Pull request merged**